website/files/models/osfstorage.py

```diff
@@ -81,10 +84,42 @@ def move_under(self, destination_parent, name=None):
             raise exceptions.FileNodeCheckedOutError()
         return super(OsfStorageFileNode, self).move_under(destination_parent, name)
 
-    def save(self):
+    def save(self):  # TODO write log
+        """
+        Uses OsfSFN save to save the node,
+            but we want to log specific activities (checkout, checkin)
+            check that via the if in the middle of the method and log the correct activity
+
+        :return:
+        """
         self.path = ''
         self.materialized_path = ''
-        return super(OsfStorageFileNode, self).save()
+        ret = super(OsfStorageFileNode, self).save()
+
+        if 'checkout' in ret and len(ret) == 1:
+            # Set parameters
+            auth = Auth(user=self.node.creator)
+
+            # is checkout vs is checkin
+            action = [lambda: NodeLog.CHECKED_IN, lambda: NodeLog.CHECKED_OUT][self.is_checked_out]()
+
+            # Log the action
+            self.node.add_log(
+                action,
+                params={
+                    'project': self.node.parent_id,
+                    'node': self.node._id,
+                    'urls': {
+                        'download': "/project/" + self.node._id + "/files/osfstorage/" + self._id + "/?action=download",
+                        'view': "/project/" + self.node._id + "/files/osfstorage/" + self._id},
+                    'path': self.materialized_path
+                },
+                auth=auth,
+            )
+
+            file_checkout_toggle.send(self.node, user=self.node.creator)
```

Comment:
If you're doing the log action here, you don't need to use signals. Emitting a signal to log the action would keep `.save()` cleaner, but signal use is also kind of ugly. I'd say to just remove the signal.